### PR TITLE
bpo-43916: Select now uses Py_TPFLAGS_DISALLOW_INSTANTIATION

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue43916

### CODE REVIEW

**Comment by erlend-aasland** [comment]
cc. @vstinner

File: Lib/test/test_select.py [hunk 1]
```diff
@@ -87,6 +87,10 @@ def fileno(self):
         a[:] = [F()] * 10
         self.assertEqual(select.select([], a, []), ([], a[:5], []))
 
+    def test_disallow_instantiation(self):
+        tp = type(select.poll())
+        self.assertRaises(TypeError, tp)
```
Comment (vstinner) [inline Lib/test/test_select.py:92]: Later we might use an helper function to also check the error message.

**Review by vstinner — APPROVED** [review]
LGTM.

File: Lib/test/test_select.py (hunk 1 above)
Comment (erlend-aasland) [reply Lib/test/test_select.py:92]: Let's add it in a separate PR. I'll update and rebase onto `main` when the helper has landed.

Comment (vstinner) [reply Lib/test/test_select.py:92]: Sure, I'm fine with with `self.assertRaises(TypeError)` for now. I mostly care about fixing https://bugs.python.org/issue43916 release blocker. Once it's no longer a release blocker, we can enhance the implementation.

**Pull request merged**